packages/@lwc/engine-core/src/framework/base-bridge-element.ts

```diff
@@ -121,9 +122,26 @@ export function HTMLBridgeElementFactory(
     const { attributeChangedCallback: superAttributeChangedCallback } = SuperClass.prototype as any;
     const { observedAttributes: superObservedAttributes = [] } = SuperClass as any;
     const descriptors: PropertyDescriptorMap = create(null);
+
+    // present a hint message so that developers are aware that they have not decorated property with @api
+    if (process.env.NODE_ENV !== 'production') {
```

Comment:
when in production don't do anything with these non decorated properties so that it stays as it was before.